src/DotNetEnv/Configuration/EnvConfigurationProvider.cs

```diff
@@ -37,9 +38,16 @@ public override void Load()
                 }
             }
 
-            // Since the Load method does not take care of cloberring, We have to check it here!
+            // Since the Load method does not take care of clobberring, We have to check it here!
             var dictionaryOption = options.ClobberExistingVars ? CreateDictionaryOption.TakeLast : CreateDictionaryOption.TakeFirst;
-            foreach (var value in values.ToDotEnvDictionary(dictionaryOption))
+            var dotEnvDictionary = values.ToDotEnvDictionary(dictionaryOption);
+
+            if (!options.ClobberExistingVars)
+                foreach (string key in Environment.GetEnvironmentVariables().Keys)
+                    if (dotEnvDictionary.ContainsKey(key))
+                        dotEnvDictionary[key] = Environment.GetEnvironmentVariable(key);
```

Comment:
Some context, if you did not look into the history of this file much: I did not write this logic (.NET configuration provider stuff). Someone else did, for purposes that I was not personally familiar with, so I just accepted as it was submitted as it was completely orthogonal to everything else. Which leaves me unsurprised that it might have skipped something important related to clobbering.

That said, this in particular seems redundant or else better logic to have elsewhere. In particular, this begs the question for me: why is `Env.Load` not already handling this, since it takes options to determine clobber/etc already?

Reply:
Said another way: maybe this is in fact the bug, but this is a very, very substantial change to the logic of this entire library: this library is a wrapper around looking up env vars, that has an in memory cache of the *overriding* env vars that come from a .env file, that may or may not be present.

This logic now copies all actual env var values into that list, if the .env has a set of keys that match actual env, rather than just skipping them or something. (I am pretty sure this logic exists somewhere already, I will find it in a moment.)

This also likely relates to your separate PR/questions about NoSetEnvVars -- we can set the env vars in the real env, that might affect other lookups, or we can only use the overrides when going through this cache. I might need to correct myself here once I refresh on actual implementations, but that is my own recollection of what I have written many years ago ;)

Reply:
https://github.com/tonerdo/dotnet-env/blob/3c7901d7e843b7572eae45c9e1f8a312dcf0c6f6/src/DotNetEnv/Env.cs#L70 is the section relating to all my above comments

Reply:
And indeed all of this logic should, in theory, be already handled here (following the logic of the above link):
https://github.com/tonerdo/dotnet-env/blob/3c7901d7e843b7572eae45c9e1f8a312dcf0c6f6/src/DotNetEnv/Parsers.cs#L22